packages/styles/forms.css

```diff
@@ -163,6 +163,10 @@ textarea.Field--has-error:focus:hover,
   padding-left: var(--space-half);
 }
 
+.Error:empty {
```

Comment:
A few points of feedback:

1. Although the PR description says "BREAKING CHANGE", I wouldn't consider this a breaking change per-say. Have we flagged similar changes to this as breaking?
2. In my experience, I typically use `:empty` as a last resort because it is a little finicky. For example, someone might do something like this:
```js
  <div className="Error">
    <span className="errorMessage">
      {error}
    </span>
  </div>
```
If `{error}` is null or empty here, it wouldn't matter because the `<span>` would make CSS think `div.Error` isn't empty. In my opinion, it would be better and more reliable if we just didn't render the div container at all when we shouldn't be:
```js
  {error && (
    <div className="Error">
      <span className="errorMessage">
        {error}
      </span>
    </div>
  )}
```

3. If we were to take this approach, though, we should probably modify `.Error` itself to be `.Error:not(:empty)`, rather than add set of overrides `.Error:empty` because we might forget some properties (for example, in order for this truly not to take up space, you probably need to add overrides for `margin-bottom` and `padding-left` as well.

Reply:
1. @schne324 probably would have better idea;
2. For the particular three components, `Select`, `TextField`, and `CheckBox`, they don't have children inside the `Error` div, but I could totally picture cases similar to your `span` example occur at some point.
3. I was taking a shortcut and I agree add conditional rendering is probably safer.